ext/rorocos/ruby_task_context.cc

```diff
@@ -64,7 +67,14 @@ struct LocalTaskContext : public RTT::TaskContext
 
     boost::int32_t __orogen_getTID() const
     {
+#ifdef __APPLE__
+        std::string thread = boost::lexical_cast<std::string>(boost::this_thread::get_id());
```

Comment:
Shouldn't we simply do that for everyone ? Does that give the same result than SYS_gettid on Linux ?